File: codegen/results/v001/qwen3.5_27b_nothink/20260408_102056/go/005_unit_test_writer/solution.go
package main

import "fmt"

// calculateDiscount calculates the total price after applying bulk discounts.
func calculateDiscount(price float64, quantity int) float64 {
	if quantity < 10 {
		return price * float64(quantity)
	} else if quantity < 50 {
		return price * float64(quantity) * 0.9
	}
	return price * float64(quantity) * 0.75
}

func main() {
	// Define test cases: {description, price, quantity, expectedResult}
	tests := []struct {
		description string
		price       float64
		quantity    int
		expected    float64
	}{
		{
			description: "Quantity 9 (No discount)",
			price:       10.0,
			quantity:    9,
			expected:    10.0 * 9.0, // 90.0
		},
		{
			description: "Quantity 10 (10% discount boundary)",
			price:       10.0,
			quantity:    10,
			expected:    10.0 * 10.0 * 0.9, // 90.0
		},
		{
			description: "Quantity 49 (10% discount boundary)",
			price:       10.0,
			quantity:    49,
			expected:    10.0 * 49.0 * 0.9, // 441.0
		},
		{
			description: "Quantity 50 (25% discount boundary)",
			price:       10.0,
			quantity:    50,
			expected:    10.0 * 50.0 * 0.75, // 375.0
		},
	}

	allPassed := true

	for _, tc := range tests {
		got := calculateDiscount(tc.price, tc.quantity)

		// Allow for small floating point precision differences
		const epsilon = 0.0001
		diff := got - tc.expected
		if diff < 0 {
			diff = -diff
		}

		if diff > epsilon {
			fmt.Printf("FAIL: %s (expected: %.2f, got: %.2f)\n", tc.description, tc.expected, got)
			allPassed = false
		} else {
			fmt.Printf("PASS: %s\n", tc.description)
		}
	}

	if allPassed {
		// Optional: Explicitly indicate all tests passed if needed, 
		// though the requirement just asked to print per test case.
	}
}

File: codegen/results/v001/qwen3.5_27b_nothink/20260408_102056/python/005_unit_test_writer/solution.py
def calculate_discount(price: float, quantity: int) -> float:
    """
    Calculate total price after bulk discount.
    quantity < 10: no discount
    quantity 10-49: 10% off
    quantity >= 50: 25% off
    Returns: price * quantity * discount_multiplier
    """
    if quantity < 10:
        return price * quantity
    elif quantity < 50:
        return price * quantity * 0.9
    else:
        return price * quantity * 0.75


def run_tests():
    # Test cases: (price, quantity, expected_total, description)
    test_cases = [
        (10.0, 9, 90.0, "Quantity 9 (below discount threshold): no discount"),
        (10.0, 10, 90.0, "Quantity 10 (lower boundary of 10% discount): 10% off"),
        (10.0, 49, 441.0, "Quantity 49 (upper boundary of 10% discount): 10% off"),
        (10.0, 50, 375.0, "Quantity 50 (lower boundary of 25% discount): 25% off"),
        (20.0, 100, 1500.0, "Quantity 100 (high quantity): 25% off"),
    ]

    for price, quantity, expected, description in test_cases:
        result = calculate_discount(price, quantity)
        # Use a small epsilon for float comparison to handle potential precision issues
        if abs(result - expected) < 1e-9:
            print(f"PASS: {description}")
        else:
            print(f"FAIL: {description} (expected: {expected}, got: {result})")


if __name__ == "__main__":
    run_tests()